app.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'library_manager'
+require_relative 'menu'
+require_relative 'person_class'
+require_relative 'student_class'
+require_relative 'teacher_class'
+require_relative 'classroom_class'
+require_relative 'book_class'
+require_relative 'rental_class'
+
+library_manager = LibraryManager.new
+menu = Menu.new(library_manager)
+
+loop do
+  menu.display_menu
+  choice = gets.chomp.to_i
+
+  if menu.valid_choice?(choice)
+    menu.handle_choice(choice)
+  else
+    puts 'Invalid choice. Please choose a valid option.'
+  end
+
+  break if choice == 8
+end
```

Comment:
- [ ] 👋 Hi there, Great work on your recent PR! I appreciate your efforts in refactoring the code. It's a positive step towards maintaining a clean and organized codebase. 🧹✨. One important thing to note is that it's highly recommended to use `main.rb` as the entry point for your application. If you choose a different name, please make sure to document it in the readme so that users can easily find the file to run. 📝👀
- [ ] I did notice that all of the options were not working as expected and were throwing errors. When refactoring, it's crucial to ensure that the functionality is maintained

![image](https://github.com/zbahati/school_library_oop/assets/98453979/a81722bc-976c-4c81-9697-022ea8144504)

####  Pay special attention to input validation and error handling to prevent unexpected input issues.